google/auth/transport/_aiohttp_requests.py

```diff
@@ -138,7 +138,12 @@ class Request(transport.Request):
     """
 
     def __init__(self, session=None):
-        self.session = None
+        # TODO: Use auto_decompress property for aiohttp 3.7+
+        if session is not None and session._auto_decompress:
+            raise ValueError(
+                "Client sessions with auto_decompress=True are not supported."
```

Comment:
why raise the error here?

Reply:
Modifying the `session` object provided by the user is inappropriate in this place, so we can't disable `auto_decompress` ourselves. Remaining options are:

- Ignore the `session` object provided by the user and leave the `self.session = None`. Log error in console.
- Raise the error to stop execution and warn user, that's something went wrong with one's `session`.

I chose the latter option because it's better to **fail early** in my opinion. A user may not notice that one's `session` object is ignored.

Reply:
Sorry I meant why auto_decompress=True are not supported. I am not familiar with aiohttp.

I think in this PR you also need to pass `auto_decompress=False` to https://github.com/googleapis/google-auth-library-python/blob/master/system_tests/system_tests_async/conftest.py#L29 to make the system test work.

Reply:
> Sorry I meant why auto_decompress=True are not supported. I am not familiar with aiohttp.
> 
> I think in this PR you also need to pass `auto_decompress=False` to https://github.com/googleapis/google-auth-library-python/blob/master/system_tests/system_tests_async/conftest.py#L29 to make the system test work.

Ah, `auto_decompress=True` is supported by `aiohttp`, but this library prefers to do decompression on its own:
https://github.com/googleapis/google-auth-library-python/blob/04c870a5bcc2c25433774a27280505d3413550e3/google/auth/transport/_aiohttp_requests.py#L80-L88

I've made required changes in `conftest.py`.

Reply:
Ah I see. Thank you! The system test is still failing. I will take a look at why it failed.